toolkit/tools/pkg/imagecustomizerlib/customizeutils.go

```diff
@@ -338,21 +339,47 @@ func enableOrDisableServices(services imagecustomizerapi.Services, imageChroot *
 func loadOrDisableModules(modules imagecustomizerapi.Modules, imageChroot *safechroot.Chroot) error {
 	var err error
 
+	loadModuleMap := make(map[string]bool)
 	for _, module := range modules.Load {
-		logger.Log.Infof("Loading kernel module (%s)", module.Name)
-		moduleFileName := module.Name + ".conf"
-		moduleFilePath := filepath.Join(imageChroot.RootDir(), "/etc/modules-load.d/", moduleFileName)
-		err = file.Write(module.Name, moduleFilePath)
-		if err != nil {
+		if err := writeModuleLoadConfiguration(module, imageChroot); err != nil {
 			return fmt.Errorf("failed to write module load configuration: %w", err)
 		}
+		loadModuleMap[module] = true
+	}
+
+	var modulesWithOptions []string
+	for module, moduleOptions := range modules.Options {
+		if !loadModuleMap[module] {
+			// load the module in options list if it is not specified in load list
+			if err := writeModuleLoadConfiguration(module, imageChroot); err != nil {
+				return fmt.Errorf("failed to write module load configuration: %w", err)
+			}
+		}
+
+		var options []string
+		logger.Log.Infof("Loading kernel module options for (%s)", module)
+		for key, value := range moduleOptions {
+			options = append(options, fmt.Sprintf("%s=%s", key, value))
+		}
+
+		data := fmt.Sprintf("options %s %s", module, strings.Join(options, " "))
+		modulesWithOptions = append(modulesWithOptions, data)
+	}
+
+	if len(modulesWithOptions) > 0 {
+		modulesWithOptions = append(modulesWithOptions, "")
+		moduleOptionsFilePath := filepath.Join(imageChroot.RootDir(), "/etc/modprobe.d/module-options.conf")
```

Comment:
It seems a little odd that for load and disable you give each modules its own config file. But for options, you combine them all into a single file.

Reply:
I think it might be better to unify on 1 file per module instead of a 1 file per operation.